### PR TITLE
yarn-plugin: Only throw if the version is not already bound

### PR DESCRIPTION
**Pull request by benjdlambert**
Ran into an issue internally that this gets called with `backstage:1.31.0` or whatever as a resolved version and then ends up throwing an error. Not sure if this is the right fix or not?

### CODE REVIEW

File: packages/yarn-plugin/src/resolver/BackstageResolver.ts [hunk 1]
```diff
@@ -44,7 +44,10 @@ export class BackstageResolver implements Resolver {
    * the version in backstage.json changes.
    */
   bindDescriptor(descriptor: Descriptor): Descriptor {
-    if (descriptor.range !== 'backstage:^') {
+    if (
+      descriptor.range !== 'backstage:^' &&
+      !descriptor.range.startsWith('backstage:')
+    ) {
```
Comment (mtlewis) [inline packages/yarn-plugin/src/resolver/BackstageResolver.ts:50]: Interesting that bindDescriptor can be called for entries in yarn.lock somehow! Curious to understand why that happens, but in any case - given this, I'd probably flip the whole thing to something like:

```typescript
if (descriptor.range === 'backstage:^') {
  return structUtils.makeDescriptor(
    descriptor,
    `${PROTOCOL}${getCurrentBackstageVersion()}`,
  );
}

return descriptor;
```

Comment (benjdlambert) [reply packages/yarn-plugin/src/resolver/BackstageResolver.ts:50]: I was even thinking we could do something like this instead: 

```ts
if (
      descriptor.range !== 'backstage:^' &&
      descriptor.range !== `${PROTOCOL}${getCurrentBackstageVersion()}`
    ) {
      throw new Error(
        `Unsupported version range "${
          descriptor.range
        }" for package ${structUtils.stringifyIdent(
          descriptor,
        )}. The backstage protocol only supports the range "backstage:^" or the current Backstage version "${PROTOCOL}${getCurrentBackstageVersion()}".`,
      );
    }

    return structUtils.makeDescriptor(
      descriptor,
      `${PROTOCOL}${getCurrentBackstageVersion()}`,
    );
```

Comment (benjdlambert) [reply packages/yarn-plugin/src/resolver/BackstageResolver.ts:50]: That way it throws if something is pointing to the wrong version?

Comment (Rugvip) [reply packages/yarn-plugin/src/resolver/BackstageResolver.ts:50]: Hmm, was kinda thinking that being able to point to a different version for an individual is a feature though?

Comment (mtlewis) [reply packages/yarn-plugin/src/resolver/BackstageResolver.ts:50]: Yeah, that was my thinking too

Comment (benjdlambert) [reply packages/yarn-plugin/src/resolver/BackstageResolver.ts]: Fixed and tests updated

**Review by Rugvip — APPROVED** [review]
LGTM, but will defer to @mtlewis on the correctness of this 😁

**Comment by benjdlambert** [comment]
Gonna merge as it was @mtlewis suggestion 😅

**Pull request merged**